internals-js/src/definitions.ts

```diff
@@ -1499,6 +1509,10 @@ export class SchemaDefinition extends SchemaElement<SchemaDefinition, Schema>  {
       const imports = extractCoreFeatureImports(url, schemaDirective);
       const core = new CoreFeature(url, args.as ?? url.name, schemaDirective, imports, args.for);
       Schema.prototype['markAsCoreSchema'].call(schema, core);
+      // We also any core features that may have been added before we saw the @link for link itself
```

Comment:
Yeah, that's the an artifact of how the code works right now. Essentially, because the abstraction is mutable, the way incomplete schema are expanded is done by taking the input and adding what's missing. So when we get an input that only has a `@link` to the federation spec, but no `@link`-to-link-itself, we end up adding that `@link`-to-link-itself to a document hat already has another existing `@link` and this code ensure that pre-existing feature is still tracker correctly.

To be clear, the resulting "final" document still end up with the `@link`-to-link as the first one on the `SchemaDefinition`, the code is just need due to way incomplete schema are auto-completed.

>  the `@core` and `@link` specs both require that the bootstrap `@link` is the first one in the document

As a minor aside, this made me notice that the current core/v0.2 spec (https://specs.apollo.dev/core/v0.2/) only ever talk about `SchemaDefinition` in the bootstrap/feature collection descriptions. Since we've making extensive doc of putting `@link` on `SchemaExtension` in our docs/examples, we should make sure to clarify their handling.